### DISCUSSION TITLE
Cypress + Laravel - file upload testing works with a PNG but not with an MP4 file

### DISCUSSION THREAD

**Discussion by Jamesking56** [cypress-io/cypress#25952] [Questions and Help]
Hi hoping I can get some help here.

I'm using Cypress 10 with Laravel 9 + InertiaJS.

I have a hidden `<input type="file" name="some-name">` and when a file is selected, it is immediately sent using Axios to a Laravel route where the file is saved to storage. This form works for the necessary file types (images and videos) when I manually test but I'm having issues getting Cypress to upload files properly.

My test does what the documentation says and uploads a file as follows: 

```
cy.get('input[name="workoutFile"]').selectFile('storage/app/example/day1-exercise1.mp4', {force: true});
```

Laravel successfully sees a valid `UploadedFile` for a .png file but for any .mp4 video, Laravel gets an invalid `UploadedFile` with no size, no conent and the wrong path.

Any ideas why mp4s don't seem to work but pngs do? Is it a bug in Cypress or something extra I need to do for mp4 files?

Cheers.

**Comment by portyu9** [comment]
Hi @Jamesking56 

This is most likely a PHP upload-size limit, not an MP4 issue in Cypress.

Laravel's `UploadedFile` becoming invalid with size `0` and an empty/invalid temporary path is what you typically see when PHP rejects the upload before Laravel receives it.

Check:

```php
$request->file('workoutFile')->getError();
$request->file('workoutFile')->getErrorMessage();
```

If the error is `UPLOAD_ERR_INI_SIZE`, increase both values in the `php.ini` used by your web server/PHP-FPM:

```ini
upload_max_filesize = 100M
post_max_size = 100M
```

Then restart PHP-FPM/web server.

If nginx is in front of Laravel, also increase:

```nginx
client_max_body_size 100M;
```

Your Cypress code itself is fine:

```js
cy.get('input[name="workoutFile"]')
  .selectFile('storage/app/example/day1-exercise1.mp4', { force: true })
```

So check the PHP upload error code first—the PNG succeeds because it is smaller, while the MP4 is likely exceeding the server upload limit.

**Reply by portyu9** [reply]
@Jamesking56 Please mark this answer as accepted if it helped, thank you.

**Reply by Jamesking56** [reply]
@portyu9 Thanks for the response. Unfortunately, I cannot verify this as I stopped using Cypress over a year ago.
